app/src/main/scala/org/alephium/explorer/ExploreHttpServer.scala

```diff
@@ -0,0 +1,88 @@
+// Copyright 2018 The Alephium Authors
+// This file is part of the alephium project.
+//
+// The library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the library. If not, see <http://www.gnu.org/licenses/>.
+
+package org.alephium.explorer
+
+import scala.collection.immutable.ArraySeq
+import scala.concurrent._
+
+import com.typesafe.scalalogging.StrictLogging
+import io.vertx.core.Vertx
+import io.vertx.core.http.{HttpMethod, HttpServer}
+import io.vertx.ext.web._
+import io.vertx.ext.web.handler.CorsHandler
+import sttp.tapir.server.vertx.VertxFutureServerInterpreter._
+
+import org.alephium.util.Service
+
+/** Stores AkkaHttp related instances created on boot-up */
+@SuppressWarnings(Array("org.wartremover.warts.NonUnitStatements"))
+class ExplorerHttpServer(host: String, port: Int, val routes: ArraySeq[Router => Route])(
+    implicit val executionContext: ExecutionContext
+) extends Service
+    with StrictLogging {
+
+  private val vertx  = Vertx.vertx()
+  private val router = Router.router(vertx)
```

Comment:
The issue #356 might be valid here? `vertx` is being created without calling `startSelfOnce`.

Reply:
nice catch, I'll have a look

Reply:
done